Add support for TF GlobalAveragePooling2D layers in conversion code.

jamesturner246 opened the pull request:
Adds TF GlobalAveragePooling2D conversion code, fixing #30.

neworderofjamie reviewed (commented):
maybe a test would be good?

jamesturner246 commented:
Made some changes and added a couple of tests. Basically solved a silly mistake, allow pool/conv size/stride to be passed as an integer or a tuple, and removed the code to have conv2d layers after global avepoo2d layers, since this doesn't make sense.

neworderofjamie commented on ml_genn/layers/avepool2d_conv2d_synapses.py:
```diff
@@ -95,14 +95,24 @@ def __init__(self, filters, pool_size, conv_size, pool_strides=None,
-        self.conv_size = conv_size
-        if pool_strides == None:
-            self.pool_strides = (pool_size[0], pool_size[1])
+        if isinstance(pool_size, int):
+            self.pool_size = (pool_size, pool_size)
+        else:
+            self.pool_size = pool_size
```
this could perhaps go in a helper function as I suspect it's fairly ubiquitous 
```python
def getParam2D(param, default=None):
    if param is None:
        if default is not None:
            return default
        else:
            raise Exception("Parameter required")
    elif isinstance(param, int):
        return (param, param)
    else:
        return param
```

jamesturner246 replied:
Done.
